### PR TITLE
RD-5789 Migrate all outstanding non-widget files to TypeScript

### PR DESCRIPTION
**Pull request by qooban**
## Description

Don't be scared about the size as most of the line-changes comes from either `package*.json` changes or API mock file (`test/jest/widgets/topology/data/ExampleData.json`).

The following non-widget files were migrated to TypeScript:
1. Frontend unit test files (leftovers from other migration tasks)
2. JS scripts from `scripts` directory
3. Webpack configuration file

Supporting changes:
1. `BlueprintPlan` type was not inline with what Cloudify REST API is providing, so it was fixed making it easier to migrate frontend unit test files

## Screenshots / Videos
N/A

## Checklist
- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests

[Build Status](https://jenkins.cloudify.co/job/Stage-UI-System-Test/4469/) - all tests passed.

## Documentation
Updated internal documentation.

### CODE REVIEW

File: test/jest/widgets/topology/DataProcessor.test.ts [hunk 1]
```diff
@@ -9,28 +10,28 @@ describe('(Widget) Topology', () => {
         it('supports `plan` from manager', () => {
             const data = {
                 data: nodecellarExampleData
-            };
+            } as unknown as ManagerData;
```
Comment (qooban) [inline test/jest/widgets/topology/DataProcessor.test.ts:13]: It was hard to have full compatibility (e.g. `string` inferred from JSON file vs. `Visibility` from the TS type)

File: test/jest/widgets/topology/NodeStatusService.test.ts [hunk 2]
```diff
@@ -7,49 +6,59 @@ describe('(Widget) Topology', () => {
     describe('NodeStatusService getColorByStatus', () => {
         it('should return color by status', () => {
             _.each(consts.nodeStatuses, s => {
-                expect(typeof getColorByStatus(s)).toBe('string', `type ${s} should have a color`);
```
Comment (qooban) [inline test/jest/widgets/topology/NodeStatusService.test.ts:10]: `toBe` does not have second argument.

File: app/widgets/common/inputs/DataTypesButton.tsx [hunk 3]
```diff
@@ -7,37 +7,25 @@ import { CancelButton, Header, Icon, Modal, Popup, Segment, Table } from '../../
 import translateInputs from './utils/translateInputs';
 import type { BlueprintPlan } from '../blueprints/BlueprintActions';
 
-type Properties = BlueprintPlan['data_types']['properties'] & {
```
Comment (qooban) [inline app/widgets/common/inputs/DataTypesButton.tsx:10]: Changes in this file are simplifications after fix in `BlueprintPlan` type.

**Comment by Vorbert-Kruk** [comment]
I'll review it tomorrow

**Comment by kubama** [comment]
> 1. `BlueprintPlan` type was not inline with what Cloudify REST API is providing, so it was fixed making it easier to migrate frontend unit test files

Are the changes in `ExampleData.json` required due to this?

**Review by kubama — APPROVED** [review]

**Comment by qooban** [comment]
@kubama 

> Are the changes in ExampleData.json required due to this?

Yep. Initially I wanted to make it matching `BlueprintPlan` type completely. But it turned out that due to https://github.com/cloudify-cosmo/cloudify-stage/pull/2423#discussion_r1097401913 it wasn't possible to have full alignment.

**Review by Vorbert-Kruk — APPROVED** [review]
LGTM 💪 
Additional kudos for simplifications 🚀

**Pull request merged**